crate_universe/tests/cargo_integration_test.rs

```diff
@@ -261,3 +261,36 @@ fn feature_generator_workspace() {
 
     assert!(!metadata["metadata"]["cargo-bazel"]["features"]["wgpu 0.14.0"].is_null());
 }
+
+#[test]
+fn feature_generator_crate_combined_features() {
+    // This test case requires network access to build pull crate metadata
+    // so that we can actually run `cargo tree`. However, RBE (and perhaps
+    // other environments) disallow or don't support this. In those cases,
+    // we just skip this test case.
+    use std::net::ToSocketAddrs;
+    if "github.com:443".to_socket_addrs().is_err() {
+        eprintln!("This test case requires network access. Skipping!");
+        return;
+    }
+
+    let runfiles = runfiles::Runfiles::create().unwrap();
+    let metadata = run(
```

Comment:
Perhaps `run` should run `cargo fetch` before calling `splice` to make sure the registry is up to date? I bet `cargo tree --locked` does not update the index? which might be a bug there?

Maybe the long term fix is to have https://github.com/bazelbuild/rules_rust/blob/0.19.0/crate_universe/src/splicing.rs#L419-L451 call `cargo fetch` or something that updates the index in the event that the function does not actually call `cargo generate-lockfile`?

Reply:
I did a quick test and `cargo tree --locked` does update the registry properly (at least on my machine) even after clearing `~/.cargo/registry`.

Reply:
One more feature I didn't really think of... [`matrix-sdk = 0.6.2` requires Rust 1.65](https://github.com/matrix-org/matrix-rust-sdk/blob/main/Cargo.toml#L20), thanks for figuring this out.